presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,532 @@
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.lang.reflect.Method;
+
+import static com.facebook.presto.geospatial.GeometryUtils.deserialize;
+import static com.facebook.presto.geospatial.GeometryUtils.serialize;
+
+public final class GeoFunctions
+{
+    private static final String LINE = "LineString";
+    private static final String MULTILINE = "MultiLineString";
+    private static final String MULTIPOINT = "MultiPoint";
+    private static final String POLYGON = "Polygon";
+    private static final String POINT = "Point";
+    private static final String GEOMETRY = "Geometry";
+
+    private GeoFunctions() {}
+
+    @Description("Construct Line Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stLine(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry line = OGCGeometry.fromText(input.toStringUtf8());
+        line.setSpatialReference(null);
+        return serialize(line);
+    }
+
+    @Description("Construct Point Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stPoint(@SqlType(StandardTypes.DOUBLE) double longitude, @SqlType(StandardTypes.DOUBLE) double latitude)
+    {
+        OGCGeometry geometry = OGCGeometry.createFromEsriGeometry(new Point(longitude, latitude), null);
+        return serialize(geometry);
+    }
+
+    @Description("Construct Polygon Geometry")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stPolygon(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        OGCGeometry polygon = OGCGeometry.fromText(input.toStringUtf8());
+        polygon.setSpatialReference(null);
+        return serialize(polygon);
+    }
+
+    @Description("Returns area of input polygon")
+    @ScalarFunction("st_area")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stArea(@SqlType(GEOMETRY) Slice input)
+    {
+        return deserialize(input).getEsriGeometry().calculateArea2D();
+    }
+
+    @Description("Returns Geometry of input wkt text")
+    @ScalarFunction("st_geometry_from_wkt")
+    @SqlType(GEOMETRY)
+    public static Slice stGeometryFromWkt(@SqlType(StandardTypes.VARCHAR) Slice input)
+    {
+        return serialize(OGCGeometry.fromText(input.toStringUtf8()));
+    }
+
+    @SqlNullable
+    @Description("Returns wtk text of input Geometry")
+    @ScalarFunction("st_geometry_to_wkt")
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice stGeometryToWkt(@SqlType(GEOMETRY) Slice input)
+    {
+        return Slices.utf8Slice(deserialize(input).asText());
+    }
+
+    @Description("Returns point that is the center of the polygon's envelope")
+    @ScalarFunction
+    @SqlType(GEOMETRY)
+    public static Slice stCentroid(@SqlType(GEOMETRY) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        String type = geometry.geometryType();
+        if (type.equals(POLYGON)) {
```

Comment:
nit: the logic of checking input's geometry type is used in multiple functions; would it make sense to move it into a helper method?

    private OGCGeometry getGeometry(Slice input, String function, String...validTypes) {
        OGCGeometry geometry = GeometryUtils.geometryFromBinary(input);
        String type = geometry.geometryType();
        for (String validType : validTypes) {
            if (type.equals(validType)) {
                return geometry;
            }
        }
        throw new IllegalArgumentException(function + " only applies to " +
                StringUtils.join(Arrays.asList(validTypes), " ,")+ ". Input type is: " + type);
    }